app/views/ShareView/index.js

```diff
@@ -1,307 +1,360 @@
-		const { server, user } = this.props;
-		const { name, description } = file;
-		const fileMessage = {
-			name,
-			description,
-			size: fileInfo.size,
-			type: fileInfo.mime,
-			store: 'Uploads',
-			path: fileInfo.path
-		};
-		if (fileInfo && rid !== '') {
-			try {
-				await RocketChat.sendFileMessage(rid, fileMessage, undefined, server, user);
-			} catch (e) {
-				log(e);
+			// Send text message
+			} else if (text.length) {
+				await RocketChat.sendMessage(room.rid, text, thread?.tmid, { id: user.id, token: user.token });
 			}
+		} catch {
+			// Do nothing
 		}
-	}
 
-	sendTextMessage = async() => {
-		const { value, rid } = this.state;
-		const { user } = this.props;
-		if (value !== '' && rid !== '') {
-			try {
-				await RocketChat.sendMessage(rid, value, undefined, user);
-			} catch (e) {
-				log(e);
-			}
+		// if it's share extension this should close
+		if (this.isShareExtension) {
+			ShareExtension.close();
 		}
 	};
 
-	renderPreview = () => {
-		const { fileInfo } = this.state;
-		const { theme } = this.props;
+	selectFile = (item) => {
+		const { text } = this.messagebox.current;
+		const { attachments, selected } = this.state;
+		const newAttachments = attachments.map((att) => {
+			if (att.path === selected.path) {
+				att.description = text;
+			}
+			return att;
+		});
+		this.setState({ attachments: newAttachments, selected: item });
+	}
 
-		const icon = fileInfo.mime.match(/image/)
-			? <Image source={{ isStatic: true, uri: fileInfo.path }} style={styles.mediaImage} />
-			: (
-				<View style={styles.mediaIconContainer}>
-					<CustomIcon name='clip' style={styles.mediaIcon} />
-				</View>
-			);
+	removeFile = (item) => {
+		const { selected, attachments } = this.state;
+		let newSelected;
+		if (item.path === selected.path) {
+			const selectedIndex = attachments.findIndex(att => att.path === selected.path);
+			// Selects the next one, if available
+			if (attachments[selectedIndex + 1]?.path) {
+				newSelected = attachments[selectedIndex + 1];
+			// If it's the last thumb, selects the previous one
+			} else {
+				newSelected = attachments[selectedIndex - 1] || {};
+			}
+		}
+		this.setState({ attachments: attachments.filter(att => att.path !== item.path), selected: newSelected ?? selected });
+	}
 
-		return (
-			<View
-				style={[
-					styles.mediaContent,
-					{
-						borderColor: themes[theme].separatorColor,
-						backgroundColor: themes[theme].auxiliaryBackground
-					}
-				]}
-			>
-				{icon}
-				<View style={styles.mediaInfo}>
-					<Text style={[styles.mediaText, { color: themes[theme].titleText }]} numberOfLines={1}>{fileInfo.name}</Text>
-					<Text style={[styles.mediaText, { color: themes[theme].titleText }]}>{this.bytesToSize(fileInfo.size)}</Text>
-				</View>
-			</View>
-		);
-	};
+	onChangeText = (text) => {
+		const { attachments } = this.state;
+		// we manage attachments on this state only when it's text only sharing
+		if (!attachments.length) {
```

Comment:
Correct me if I'm wrong, but I don't guess so we will call this function when this component has some attachment.